utilities/xmlrpcpp/src/XmlRpcValue.cpp

```diff
@@ -610,7 +610,14 @@ namespace XmlRpc {
       default:           break;
       case TypeBoolean:  os << _value.asBool; break;
       case TypeInt:      os << _value.asInt; break;
-      case TypeDouble:   os << _value.asDouble; break;
+      case TypeDouble:
+        {
+          char buf[100]; // Should be long enough
```

Comment:
> Should be long enough

The integer part of a double can be much larger - at least 309 digits. And there can be a `-` sign in front. The problem is that the format string can extend the size with decimal digits, e.g. `%.999f`.

So I am not sure that `100` is a reasonable upper limit. Not sure if we should just use a higher boundary or if we should do this dynamically:
* determine the necessary buffer size,
* allocate that buffer dynamically,
* printf the double value,
* stream it into the out stream
* deallocate the temporary buffer.

Reply:
I suggest an opportunistic approach. The [return value of snprintf](http://www.cplusplus.com/reference/cstdio/snprintf/#return) tells the required size of the buffer. So it should be possible to have the 100-byte (or whatever) buffer statically initialized, and if the formatted string fits inside, go on with this static buffer. If the buffer is too small, then dynamic allocation should create the buffer of the size snprintf tells.

This would result in a small slowdown for long strings, but would avoid dynamic allocation for more common cases.

Reply:
That sounds like the best of both :+1: